### PR TITLE
[SPARK-42757][CONNECT] Implement textFile for DataFrameReader

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to implement textFile for DataFrameReader.

### Why are the changes needed?
API coverage.

### Does this PR introduce _any_ user-facing change?
New method.

### How was this patch tested?
Add new UT.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @hvanhovell

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 1]
```diff
@@ -531,10 +531,7 @@ class DataFrameReader private[sql] (sparkSession: SparkSession) extends Logging
    */
   @scala.annotation.varargs
   def textFile(paths: String*): Dataset[String] = {
-    // scalastyle:off throwerror
-    // TODO: this method can be supported and should be included in the client API.
-    throw new NotImplementedError()
-    // scalastyle:on throwerror
```
Comment (HyukjinKwon) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:537]: should we add an assert on schema to match with the regular Spark path?

**Comment by hvanhovell** [comment]
Thanks for doing this. Small comment on the implementation.

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala (hunk 1 above)
Comment (hvanhovell) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:537]: What does that mean?

Comment (HyukjinKwon) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:537]: Meaning that adding:

```diff
    def textFile(paths: String*): Dataset[String] = {
    + assertNoSpecifiedSchema("textFile")
      text(paths : _*).select("value").as[String](sparkSession.implicits.newStringEncoder)
    }
```

check.

Comment (panbingkun) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:537]: Done

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 2]
```diff
@@ -531,10 +531,7 @@ class DataFrameReader private[sql] (sparkSession: SparkSession) extends Logging
...
+    text(paths: _*).select("value").as[String](sparkSession.implicits.newStringEncoder)
```
[the remark this answers is not in the document]
Comment (panbingkun) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: Done

**Review by hvanhovell — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.4.

**Pull request closed without merging**